src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -233,12 +241,18 @@ private static bool ContainsUpperCaseLetter(string pattern)
                 // d) If the part was not entirely lowercase, then check if it is contained in the
                 //    candidate in a case *sensitive* manner. If so, return that there was a substring
                 //    match.
-                var caseSensitiveIndex = _compareInfo.IndexOf(candidate, patternChunk.Text);
-                if (caseSensitiveIndex > 0)
+
+                // We could only get a case sensitive match if there was a case insensitive match.
+                // don't bother searching if the first case insensitive match failed.
+                if (caseInsensitiveIndex > 0)
```

Comment:
this avoids a second IndexOf call when it would always fail.  IndexOf is 60% of the search:

![image](https://user-images.githubusercontent.com/4564579/27765957-534d5c32-5e75-11e7-82d1-1cbe8dfbcacf.png)